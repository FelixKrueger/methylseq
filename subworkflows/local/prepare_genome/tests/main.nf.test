nextflow_workflow {

    name "Test Workflow PREPARE_GENOME"
    script "../main.nf"
    workflow "PREPARE_GENOME"

    test("BISMARK - All Indexes supplied") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = file(params.pipelines_testdata_base_path + 'reference/genome.fa.fai', checkIfExists: true) // fasta_index
                input[2] = file(params.pipelines_testdata_base_path + 'reference/Bowtie2_Index.tar.gz',  checkIfExists: true) // bismark_index
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

    test("BISMARK - Generate Fasta Index") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = []  // fasta_index
                input[2] = file(params.pipelines_testdata_base_path + 'reference/Bowtie2_Index.tar.gz',  checkIfExists: true) // bismark_index
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

    test("BISMARK - Generate Fasta Index & Bowtie Index") {

        when {
            params {
                aligner = "bismark"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = [] // fasta_index
                input[2] = [] // bismark_index
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bismark_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

    test("BWAMETH - All Indexes supplied") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = file(params.pipelines_testdata_base_path + 'reference/genome.fa.fai', checkIfExists: true) // fasta_index
                input[2] = [] // bismark_index
                input[3] = file(params.pipelines_testdata_base_path + 'reference/Bwameth_Index.tar.gz',  checkIfExists: true) // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

    test("BWAMETH - Generate Fasta Index") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = []  // fasta_index
                input[2] = [] // bismark_index
                input[3] = file(params.pipelines_testdata_base_path + 'reference/Bwameth_Index.tar.gz',  checkIfExists: true) // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

    test("BWAMETH - Generate Fasta Index & bwameth Index") {

        when {
            params {
                aligner = "bwameth"
            }
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true) // fasta
                input[1] = [] // fasta_index
                input[2] = [] // bismark_index
                input[3] = [] // bwameth_index
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fasta,
                    workflow.out.fasta_index,
                    workflow.out.bwameth_index,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

}
