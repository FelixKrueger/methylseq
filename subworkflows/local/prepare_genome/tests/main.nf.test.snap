{
    "BWAMETH - All Indexes supplied": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:11:39.601007949"
    },
    "BISMARK - Generate Fasta Index": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/Bowtie2_Index.tar.gz"
            ],
            [
                "versions.yml:md5,67124f8c3356bccd688e07c65a52e48a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:13:17.406358667"
    },
    "BWAMETH - Generate Fasta Index & bwameth Index": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
            ],
            [
                [
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                    "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                    "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                    "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                    "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                    "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                    "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                ]
            ],
            [
                "versions.yml:md5,67124f8c3356bccd688e07c65a52e48a",
                "versions.yml:md5,cb3f3b6e54f9a7b3b6d4f5f875e89442"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:12:28.477635099"
    },
    "BWAMETH - Generate Fasta Index": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
            ],
            [
                "versions.yml:md5,67124f8c3356bccd688e07c65a52e48a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:11:57.30233052"
    },
    "BISMARK - All Indexes supplied": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa.fai"
            ],
            [
                "/nf-core/test-datasets/raw/methylseq/reference/Bowtie2_Index.tar.gz"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:12:48.994204422"
    },
    "BISMARK - Generate Fasta Index & Bowtie Index": {
        "content": [
            [
                "/nf-core/test-datasets/raw/methylseq/reference/genome.fa"
            ],
            [
                "genome.fa.fai:md5,84f5b44c25877ac58b154706cc8bc451"
            ],
            [
                [
                    [
                        [
                            "BS_CT.1.bt2:md5,d53bef7d951c7d08e327944581c911eb",
                            "BS_CT.2.bt2:md5,752a03fc364115fc910e5a28ae154382",
                            "BS_CT.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                            "BS_CT.4.bt2:md5,02b3d1855a67fd9cbb4c411406a22fde",
                            "BS_CT.rev.1.bt2:md5,52ae603fff473b3b25eecd36c533edb9",
                            "BS_CT.rev.2.bt2:md5,0e290e05f7dae065cae54e29ed97afe5",
                            "genome_mfa.CT_conversion.fa:md5,4dd181b65f1f3549f4132c9a3759eee8"
                        ],
                        [
                            "BS_GA.1.bt2:md5,9aff51b1712758b891b0c427a988977f",
                            "BS_GA.2.bt2:md5,579241d95941538b2e75bbdb6cfaa73d",
                            "BS_GA.3.bt2:md5,b41e72eefad74e8ede411c78a5bd5dee",
                            "BS_GA.4.bt2:md5,6b64fa496ffa2af2af235e1c87f06a25",
                            "BS_GA.rev.1.bt2:md5,dabf2d6f80ba7e8fc96dfdc203f6bbfd",
                            "BS_GA.rev.2.bt2:md5,fac2f8b39b0e15dec7aa79a0ecf12b49",
                            "genome_mfa.GA_conversion.fa:md5,3dee732bc344fd620c3a7322f52c6a5f"
                        ]
                    ],
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de"
                ]
            ],
            [
                "versions.yml:md5,48a4baa99421c3dd839946dd1af8d7aa",
                "versions.yml:md5,67124f8c3356bccd688e07c65a52e48a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-08T07:13:52.673839987"
    }
}